pkg/synchronizer/kubernetes/sync_server.go

```diff
@@ -236,3 +243,57 @@ func (sync *KubeSynchronizer) GetRemoteClient() client.Client {
 func (sync *KubeSynchronizer) GetRemoteNonCachedClient() client.Client {
 	return sync.RemoteNonCachedClient
 }
+
+// startCleanup looks up all the subscriptionstatuses
+// For each subscriptionstatus that doesn't have a subscription
+// Delete the resources listed inside the subscriptionstatus
+// If all the resources are deleted successfully then delete the subscriptionstatus
+func startCleanup(synchronizer *KubeSynchronizer) {
```

Comment:
This solution works, but I would like to propose an additional improvement. Currently, we have several bugs (BZs) that indicate that the workload is not being deleted until after an hour from the node restart. I have added a comment here to explain this issue [here](https://bugzilla.redhat.com/show_bug.cgi?id=2155472#c10)

So my suggestion is to use an exponential delay that increases up to a maximum of one hour.

Reply:
@xiangjingli what do you think about @BenamarMk 's suggestion?

Reply:
Yes, it is good suggestion. Mike will implement the exponential backoff wait mechanism

Reply:
@BenamarMk done as suggested in https://github.com/open-cluster-management-io/multicloud-operators-subscription/pull/308

Reply:
Thank you @mikeshng